core/service/src/vault/keychain/mod.rs

```diff
@@ -74,6 +78,30 @@ pub enum EnvelopeStore {
     OperatorBackupOutput(BackupCiphertext),
 }
 
+#[derive(EnumTryAs, Serialize, Deserialize, Debug)]
+pub enum RootKeyMeasurements {
+    AwsKms {
+        key_origin: String,
+        key_policy: IAMPolicy,
+    },
+    SecretSharing {},
+}
+
+pub fn verify_root_key_measurements(
```

Comment:
could you add a unit test for this function as it seems to be an important component for security. perhaps with a malicious `make_root_key_policy`